caffe2/python/rnn_cell.py

```diff
@@ -234,6 +234,71 @@ def create_states(self, model):
         ]
 
 
+# based on http://pytorch.org/docs/master/nn.html#torch.nn.RNNCell
+class ElmanCell(RNNCell):
+    def __init__(
+        self,
+        input_size,
+        hidden_size,
+        activation,
+        forget_bias,
+        memory_optimization,
+        drop_states=False,
+        initializer=None,
+        **kwargs
+    ):
+        super(ElmanCell, self).__init__(**kwargs)
+        self.input_size = input_size
+        self.hidden_size = hidden_size
+        self.activation = activation
+
+
+    def _apply(
+        self,
+        model,
+        input_t,
+        seq_lengths,
+        states,
+        timestep,
+        extra_inputs=None,
+    ):
+        hidden_t_prev = states[0]
+
+        gates_t = self.scope('gates_t')
+        hidden_t = self.scope('hidden_t')
+
+        brew.fc(
+            model,
+            hidden_t_prev,
+            gates_t,
+            dim_in=self.hidden_size,
+            dim_out=self.hidden_size,
+            axis=2,
+        )
+        brew.sum(model, [gates_t, input_t], gates_t)
+        if self.activation == 'tanh':
```

Comment:
throw if non of the expected ones? and check in the constructor?

Reply:
done